pyanaconda/network.py

```diff
@@ -1160,13 +1160,17 @@ def ibftIface():
 
 def hostname_from_cmdline(cmdline):
     # legacy hostname= option
-    hostname = flags.cmdline.get('hostname', "")
+    hostname = cmdline.get('hostname', "")
     # ip= option
-    ipopts = flags.cmdline.get('ip')
+    ipopts = cmdline.get('ip')
     # Example (2 options):
     # ens3:dhcp 10.34.102.244::10.34.102.54:255.255.255.0:myhostname:ens9:none
     if ipopts:
         for ipopt in ipopts.split(" "):
+            if ipopt.startswith("["):
+                # Replace ipv6 addresses with empty string, example of ipv6 config:
+                # [fd00:10:100::84:5]::[fd00:10:100::86:49]:80:myhostname:ens9:none
+                ipopt = re.sub(r'\[[^\]]+]', '[]', ipopt)
```

Comment:
Wouldn't it be better to have this regular expression compiled before in an init?

Reply:
For consistency and better readability, I'll do it.